common/macresman.cpp

```diff
@@ -82,15 +84,16 @@ void MacResManager::close() {
 	delete[] _resLists; _resLists = nullptr;
 	delete[] _resTypes; _resTypes = nullptr;
 	delete _stream; _stream = nullptr;
+	delete _dataForkStream; _dataForkStream = nullptr;
 	_resMap.numTypes = 0;
 }
 
 bool MacResManager::hasDataFork() const {
-	return !_baseFileName.empty();
+	return !_baseFileName.empty() && (_mode == kResForkMacBinary || _dataForkStream != nullptr);
```

Comment:
What if `_mode == kResForkMacBinary` but there is no data fork, only resource?

Reply:
MacBinary always has a data fork just like real Mac file. It might be zero-sized but it's always there. Just like Unix regular file always has contents. It might be zero-sized but it's there. Ideally we should kill hasDataFork altogether since we speak about Mac files in this context and they always have data fork